x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/routes/entity_calculation.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/core/server';
+import { buildSiemResponse } from '@kbn/lists-plugin/server/routes/utils';
+import { transformError } from '@kbn/securitysolution-es-utils';
+
+import { APP_ID, RISK_SCORE_ENTITY_CALCULATION_URL } from '../../../../../common/constants';
+import type { AfterKeys } from '../../../../../common/entity_analytics/risk_engine';
+import { riskScoreEntityCalculationRequestSchema } from '../../../../../common/entity_analytics/risk_engine/risk_score_entity_calculation/request_schema';
+import { buildRouteValidation } from '../../../../utils/build_validation/route_validation';
+import { getRiskInputsIndex } from '../get_risk_inputs_index';
+import type { CalculateAndPersistScoresResponse, EntityAnalyticsRoutesDeps } from '../../types';
+import { RiskScoreAuditActions } from '../audit';
+import { AUDIT_CATEGORY, AUDIT_OUTCOME, AUDIT_TYPE } from '../../audit';
+import { convertRangeToISO } from '../tasks/helpers';
+import { buildRiskScoreServiceForRequest } from './helpers';
+import { getFieldForIdentifier } from '../helpers';
+import { withRiskEnginePrivilegeCheck } from '../../risk_engine/risk_engine_privileges';
+
+export const riskScoreEntityCalculationRoute = (
+  router: EntityAnalyticsRoutesDeps['router'],
+  getStartServices: EntityAnalyticsRoutesDeps['getStartServices'],
+  logger: Logger
+) => {
+  router.versioned
+    .post({
+      path: RISK_SCORE_ENTITY_CALCULATION_URL,
+      access: 'internal',
+      options: {
+        tags: ['access:securitySolution', `access:${APP_ID}-entity-analytics`],
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: { body: buildRouteValidation(riskScoreEntityCalculationRequestSchema) },
```

Comment:
we can use the zod generator and the types generated from your OpenAPI spec here, see the asset criticality upsert route for an example

https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/routes/upsert.ts#L34